### PR TITLE
Implemented GetDimensions module

### PR DESCRIPTION
**Pull request by Smit1603**
- This pull request will resolve issue #19 
-  I have implemented GetDimensions module and tested it successfully
-  Kindly review and let me know whether any further changes are required or not

### CODE REVIEW

File: include/clang/Interpreter/InterOp.h [hunk 1]
```diff
@@ -190,6 +190,7 @@ namespace InterOp {
   std::vector<std::string> GetAllCppNames(TCppScope_t scope);
 
   void DumpScope(TCppScope_t scope);
+  std::vector<TCppIndex_t> GetDimensions(TCppType_t OpaqueQualType);
```
[the remark this answers is not in the document]
Comment (sudo-panda) [reply include/clang/Interpreter/InterOp.h:193]:
```suggestion
  std::vector<TCppIndex_t> GetDimensions(TCppType_t type);
```

Naming it `OpaqueQualType` exposes the underlying API being used. I think we should stick to generic names. @vgvassilev what's your thought on this?

**Review by sudo-panda — CHANGES_REQUESTED** [review]
I have requested some minor changes otherwise the PR looks good!

File: include/clang/Interpreter/InterOp.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (vgvassilev) [reply include/clang/Interpreter/InterOp.h:193]: I would agree also for consistency reasons.

**Comment by Smit1603** [comment]
- I have made necessary changes as mentioned by @vgvassilev and @sudo-panda 
- Kindly review changes and pls let me know whether any further improvement is required or not
- Changes made : 

- - Renamed OpaqueQualType to Type
- - Modified types wherever necessary

File: include/clang/Interpreter/InterOp.h [hunk 2]
```diff
@@ -190,6 +190,8 @@ namespace InterOp {
   std::vector<std::string> GetAllCppNames(TCppScope_t scope);
 
   void DumpScope(TCppScope_t scope);
+
+  std::vector<TCppIndex_t> GetDimensions(TCppType_t Type);
```
Comment (sudo-panda) [inline include/clang/Interpreter/InterOp.h:194]:
```suggestion
  std::vector<TCppIndex_t> GetDimensions(TCppType_t type);
```

File: lib/Interpreter/InterOp.cpp [hunk 3]
```diff
@@ -2164,5 +2164,24 @@ namespace InterOp {
 
     return names;
   }
+
+  std::vector<TCppIndex_t> GetDimensions(TCppType_t Type)
+  {
+    QualType Qual = QualType::getFromOpaquePtr(Type);
```
Comment (sudo-panda) [inline lib/Interpreter/InterOp.cpp:2170]:
```suggestion
  std::vector<TCppIndex_t> GetDimensions(TCppType_t type)
  {
    QualType Qual = QualType::getFromOpaquePtr(type);
```

**Review by sudo-panda — CHANGES_REQUESTED** [review]
For consistency, please use `snake_case` for the arguments of functions in the API.

**Comment by Smit1603** [comment]
- Sorry was unaware about naming format used
- Have changed to snake_case format
- Please review

**Comment by Smit1603** [comment]
- @vgvassilev , I hope your requirement has been met.
- If any further changes are required pls let me know

**Comment by sudo-panda** [comment]
@Smit1603 instead of merging, rebase your branch on the master branch.

**Comment by Smit1603** [comment]
- @sudo-panda I have made some changes
- Were this the changes recommended by you and is it now okay??

**Review by sudo-panda — APPROVED** [review]
LGTM!

**Comment by sudo-panda** [comment]
@vgvassilev can you run the CI before merging?

**Pull request merged**